viewing_party/party.py

```diff
@@ -1,23 +1,131 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {
+        "title": title, 
+        "genre": genre,
+        "rating": rating
+    }
+
+    if None in movie.values():
+        return None
+
+    else:
+        return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    for movie in list(user_data["watchlist"]):
+        if movie["title"] == movie_title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    if not user_data["watched"]:
+        return 0
 
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+
+    average = sum / len(user_data["watched"])
+    return average
+
+def get_most_watched_genre(user_data):
+    genre_count = {}
+    genre_list = []
+# I guess this works.. has functionality to deal with
+# any list of genres instead of hardcoding counters for each.
+# Maybe better with a helper function called get_genre_list()?
+    for movie in user_data['watched']:
+        count = 0
+# Add to running list of genres
+        if movie["genre"] not in genre_count:
+            genre_count[movie["genre"]] = 1
+        else:
+            genre_count[movie["genre"]] += 1
+# Iterate over the list to update a counter
+        # for genre in genre_list:
+        #     if genre == movie["genre"]:
+        #         count += 1
+# Add a key with the genre name and value with count to a dict
+        # genre_count[movie["genre"]] = count
+# Iterate over the dict and return the genre with highest count
+    for genre in genre_count:
+        if genre_count[genre] == max(genre_count.values()):

```

Comment:
Using `max()` here will iterate over the entire contents of the genre count values every loop. Consider calculating the max once before entering the loop and reusing it.